### PR TITLE
User/lav kumar/task 2

### PR DESCRIPTION
**Pull request by Lavkumar-tezo**

### CODE REVIEW

File: HTML/emp-directory.html [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>Tezo Employee Directory</title>
+  <link rel="icon" type="image/x-icon" href="../assets/images/tezo-logo-min.png">
+  <link rel="stylesheet" href="../CSS/employee.css" />
+  <link rel="stylesheet" href="../CSS/emp-directory.css" />
+  <link rel="stylesheet" href="../CSS/add-employee.css" />
+  <script src="../js/employee.js" defer></script>
+  <script src="../js/extension.js" defer></script>
```
Comment (Shashank-Reddy-N) [inline HTML/emp-directory.html]: Create a folder called "common" or "shared" inside js and add common js into it.

File: HTML/emp-directory.html [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+              </div>
+              <div class="emp-contact-container d-grid">
+                <div class="emp-email">
+                  <label class="form-label" for="new-emp-email">Email</label><br>
+                  <input type="email" placeholder=" Joe.j@tezo.com" id="new-emp-email" name="email" />
+                  <span></span>
+                </div>
+                <div class="emp-phone">
+                  <label class="form-label" for="new-emp-mobile">Mobile Number</label><br>
+                  <input type="number" maxlength="10" minlength="10" name="mobile" placeholder=" 0000000000" />
```
Comment (Shashank-Reddy-N) [inline HTML/emp-directory.html]: Remove the extra space in the beginning of placeholder value.

File: HTML/role-details.html [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+          </div>
+  
+          <div class="side-sec w-100 role-manage-sec">
+            <div class="side-sec-heading p-5">
+              <div class="md-heading">Role / User Management</div>
+              <div class="sm-heading">Role</div>
+            </div>
+            <div class="side-section-container">
+              <a href="../HTML/roles.html" class="anchor">
+                <div class="side-section-card sidebar-active-card flex-container jus-content-btw">
```
Comment (Shashank-Reddy-N) [inline HTML/role-details.html]: Give appropriate class names

File: HTML/emp-directory.html [hunk 4]
```diff
@@ -0,0 +1,550 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>Tezo Employee Directory</title>
+  <link rel="icon" type="image/x-icon" href="../assets/images/tezo-logo-min.png">
+  <link rel="stylesheet" href="../CSS/employee.css" />
```
Comment (Shashank-Reddy-N) [inline HTML/emp-directory.html]: Create a folder called "common" or "shared" inside css and add common css into it.

File: CSS/add-employee.css [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    --green-800:#008000;
+    --blue-100:#B0B9C0;
+    --white: #ffffff;
+    --white-900:#ffffff;
+    --red-400:#FDE9E9;
+    --red-800:#F44848;
+    --red-100:#FDE6E6;
+    --grey-100:#F7F8F9;
+    --grey-400:#A7A9AB;
+    --slectBg:rgba(244,245,247,255);
```
Comment (Shashank-Reddy-N) [inline CSS/add-employee.css]: for variables prefix should be "--" and suffix should follow kebab-case.
Change --slectBg to the actual color name.

File: CSS/add-employee.css [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    --activInput:rgba(142,191,255,255);
```
Comment (Shashank-Reddy-N) [inline CSS/add-employee.css]: change --activInput to its respective color name

File: CSS/add-employee.css [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    border:2px solid rgba(227,229,233,255);
+    border-radius: 2px;
+    padding-left: 0.25rem;
+}
+.employee-image-upload, .personal-info-container, .employement-info-container{
+    width: 70%;
+}
+.employee-image-upload{
+    gap:1.2rem;
+    align-items: center;
```
Comment (Shashank-Reddy-N) [inline CSS/add-employee.css:58]: maintain a separate class for align-items center

File: CSS/add-employee.css [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    width:100%;
+}
+#new-emp-dob,#edit-empl-dob{
+    border: 0;
+    outline: 0;
+    padding: 0;
+    height: 100%;
+}
+.dob-input{
+    border:2px solid rgba(227,229,233,255);
```
Comment (Shashank-Reddy-N) [inline CSS/add-employee.css]: Maintain a variable for color

File: js/employee-form.js [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+      }
+      else if (element.type == 'email') {
+        let email = element.value.toLowerCase();
+        if (!email.endsWith("tezo.com")) {
+          showValidInput(element, `&#9888;Email should be of tezo`,flag)
+          check = 0;
+        }
+      }
+      if (element.type != "file") {
+        if (element.name != "dob") {
```
Comment (Shashank-Reddy-N) [inline js/employee-form.js]: Try to minimize using if conditions. Try using && or || operators correspondingly.

File: js/employee.js [hunk 10]
```diff
@@ -0,0 +1,213 @@
+var sideBarChange = 0;
+var appUpdate = 0;
+var employeeList = [
```
Comment (Shashank-Reddy-N) [inline js/employee.js]: Have 2 to 3 objects only.

File: js/employee.js [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+        }
+        if (appUpdate != 1)
+            document.getElementsByClassName('app-update-container')[0].style.display = "block";
+        sideBarChange = 0;
+    }
+}
+
+function dismissBtn() {
+    document.getElementsByClassName("app-update-container")[0].style.display = "none";
+    appUpdate = 1;
```
Comment (Shashank-Reddy-N) [inline js/employee.js]: What is appUpdate? Rename it to appropriate name.

File: js/role-details.js [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+    let employeeContainer=createNewElement('div','w-100','role-details')
+    let departmentIcon=createNewElement('div','dept-icon','d-flex')
+    let deptImage=createNewElementWithAttr('img',['src',"../assets/icons/emp-id.svg"],['alt','department-id'])
+    departmentIcon.appendChild(deptImage)
+    let departmentId=createNewElement('div','emp-office-detail')
+    departmentId.textContent = employee.empNo;
+    let departmentDetails=createNewElement('div',"role-department", "d-flex", "w-100", "jus-content-start")
+    departmentDetails.appendChild(departmentIcon);
+    departmentDetails.appendChild(departmentId);
+    employeeContainer.appendChild(departmentDetails);
```
Comment (Shashank-Reddy-N) [inline js/role-details.js]: After creating of all the elements, try appending all the childs using for loop

File: js/roles.js [hunk 13]
```diff
@@ -0,0 +1,492 @@
...
+  let roleDetails=createNewElement('div',"w-100","role-details","flex-container","flex-col")
+  let roleDepartment=createNewElement('div',"role-department","d-flex","jus-content-btw","w-100")
+  let deptIcon=createNewElement('div','dept-icon','d-flex')
+  let deptImg=createNewElementWithAttr('img',['src',"../assets/icons/emp-id.svg"],['alt','department-icon'])
+  const deptLabel = document.createTextNode("Department");
+  deptIcon.appendChild(deptImg);
+  deptIcon.appendChild(deptLabel);
+  let roleDeptName=createNewElement('div','role-dept-name')
+  roleDeptName.textContent = data.dept;
+  roleDepartment.appendChild(deptIcon);
```
Comment (Shashank-Reddy-N) [inline js/roles.js]: After creating all the elements, try to maintain a list and add it to the dom using for loop

Comment (Lavkumar-tezo) [reply js/roles.js]: All resolved

**Pull request merged**